tests/datavolume_test.go

```diff
@@ -2305,6 +2308,190 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
+
+		updatePV := func(updateFunc func(*v1.PersistentVolume)) {
+			pv, err := f.K8sClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			updateFunc(pv)
+			_, err = f.K8sClient.CoreV1().PersistentVolumes().Update(context.TODO(), pv, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		createPV := func(scName string) {
+			dv := utils.NewDataVolumeForBlankRawImage("blank-source", "106Mi")
+			dv, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dv)
+			Expect(err).ToNot(HaveOccurred())
+			f.ForceBindPvcIfDvIsWaitForFirstConsumer(dv)
+
+			err = utils.WaitForDataVolumePhase(f, f.Namespace.Name, cdiv1.Succeeded, dv.Name)
+			Expect(err).ToNot(HaveOccurred())
+
+			pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dv.Namespace).Get(context.TODO(), dv.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			pvName = pvc.Spec.VolumeName
+
+			By("retaining pv")
+			updatePV(func(pv *v1.PersistentVolume) {
+				pv.Spec.PersistentVolumeReclaimPolicy = v1.PersistentVolumeReclaimRetain
+			})
+
+			err = f.K8sClient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Delete(context.TODO(), pvc.Name, metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			updatePV(func(pv *v1.PersistentVolume) {
+				pv.Spec.StorageClassName = scName
+				pv.Spec.ClaimRef = nil
+			})
+		}
+
+		createStorageClass := func(scName string) {
+			var err error
+			By(fmt.Sprintf("creating storage class %s", scName))
+			sc := utils.DefaultStorageClass.DeepCopy()
+			sc.Name = scName
+			sc.ResourceVersion = ""
+			sc.Annotations[controller.AnnDefaultStorageClass] = "false"
+			testSc, err = f.K8sClient.StorageV1().StorageClasses().Create(context.TODO(), sc, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		AfterEach(func() {
+			if testSc != nil {
+				err := f.K8sClient.StorageV1().StorageClasses().Delete(context.TODO(), testScName, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				testSc = nil
+			}
+
+			if pvName != "" {
+				updatePV(func(pv *v1.PersistentVolume) {
+					pv.Spec.PersistentVolumeReclaimPolicy = v1.PersistentVolumeReclaimDelete
+				})
+				pvName = ""
+			}
+		})
+
+		table.DescribeTable("import DV using StorageSpec without AccessModes, PVC is created only when", func(scName string, scFunc func(string)) {
+			By(fmt.Sprintf("verifying no storage class %s", testScName))
+			_, err := f.K8sClient.StorageV1().StorageClasses().Get(context.TODO(), scName, metav1.GetOptions{})
+			Expect(err).To(HaveOccurred())
+
+			By(fmt.Sprintf("creating new datavolume %s with StorageClassName %s", dataVolumeName, scName))
+			dataVolume := utils.NewDataVolumeWithHTTPImportAndStorageSpec(
+				dataVolumeName, "100Mi", fmt.Sprintf(utils.TinyCoreQcow2URL, f.CdiInstallNs))
+			dataVolume.Spec.Storage.StorageClassName = pointer.String(scName)
+			dataVolume.Spec.Storage.AccessModes = nil
```

Comment:
I know you are being explicit here with no access modes, but no reason for this line as it will be nil if not specified.

Reply:
`NewDataVolumeWithHTTPImportAndStorageSpec()` sets it to `ReadWriteOnce`, so we need to set it to nil here.